test/test_other.py

```diff
@@ -8086,6 +8086,30 @@ def test_binaryen_warn_mem(self):
     self.run_process([EMCC, test_file('hello_world.c'), '-sINITIAL_MEMORY=' + str(16 * 1024 * 1024), '--pre-js', 'pre.js', '-sALLOW_MEMORY_GROWTH', '-sWASM_ASYNC_COMPILATION=0', '-sIMPORTED_MEMORY'])
     self.assertContained('hello, world!', self.run_js('a.out.js'))
 
+  def test_initial_heap(self):
+    for args, expected_initial_heap in [
+        ([], 16 * 1024 * 1024),         # Default behavior: 16MB initial heap
+        (['-sINITIAL_MEMORY=40MB'], 0), # Backwards compatibility: no initial heap (we can't tell if it'll fit)
+        (['-sMAXIMUM_MEMORY=40MB'], 0), # Backwards compatibility: no initial heap (we can't tell if it'll fit)
+        (['-sINITIAL_HEAP=64KB'], 64 * 1024), # Explicitly set initial heap is passed
+        (['-sINITIAL_HEAP=128KB', '-sINITIAL_MEMORY=20MB', '-sMAXIMUM_MEMORY=40MB'], 128 * 1024),
+        (['-sINITIAL_HEAP=10MB', '-sINITIAL_MEMORY=10MB'], -1), # Not enough space for stack
+        (['-sINITIAL_HEAP=5MB', '-sMAXIMUM_MEMORY=5MB'], -1), # Not enough space for stack
+      ]:
+      cmd = [EMCC, test_file('hello_world.c'), "-v"] + args
+      print(' '.join(cmd))
+
+      if expected_initial_heap < 0:
+        out = self.expect_fail(cmd)
+        self.assertContained('wasm-ld: error:', out)
```

Comment:
What is the error that wasm-ld produces here?

Reply:
There are two distinct errors (for minimum/maximum):

https://github.com/llvm/llvm-project/blob/183c6c5d9ec892fcb7021991c579549b9e22ce3a/lld/test/wasm/initial-heap.test#L26-L27

It's intentional that there is no testing of the exact error string, since that is upstream functionality.